### ISSUE TITLE
The Long Term Data page is not populating with data.

### ISSUE THREAD

**Issue by stan-qaz** [pi-hole/pi-hole#2476]
**In raising this issue, I confirm the following:** `{please fill the checkboxes, e.g: [X]}`

- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [x] The issue I am reporting can be *replicated*.
- [x] The issue I am reporting isn't a duplicate (see [FAQs](https://github.com/pi-hole/pi-hole/wiki/FAQs), [closed issues](https://github.com/pi-hole/pi-hole/issues?utf8=%E2%9C%93&q=is%3Aissue%20is%3Aclosed%20), and [open issues](https://github.com/pi-hole/pi-hole/issues)).

**How familiar are you with the the source code relevant to this issue?:**
1
---
**Expected behaviour:**

The Long Term Data page should populate with data.

**Actual behaviour:**

The Long Term Data page is not populating with data.
The FTL log was showing readonly errors:

pi@pi-3b-3:~ $ tail -f  /var/log/pihole-FTL.log
[2018-10-21 18:45:03.056] Listening on port 4711 for incoming IPv6 telnet connections
[2018-10-21 18:45:03.057] Listening on Unix socket
[2018-10-21 18:45:03.058] Compiled 0 Regex filters and 2 whitelisted domains in 0.1 msec (0 errors)
[2018-10-21 18:45:03.058] /etc/pihole/black.list: parsed 0 domains (took 0.0 ms)
[2018-10-21 18:45:03.872] /etc/pihole/gravity.list: parsed 129535 domains (took 814.0 ms)
[2018-10-21 18:46:01.460] save_to_DB() - SQL error (8): attempt to write a readonly database
[2018-10-21 18:46:01.464] save_to_DB() - SQL error (8): attempt to write a readonly database
[2018-10-21 18:46:01.464] save_to_DB() - SQL error (8): attempt to write a readonly database
[2018-10-21 18:46:01.465] save_to_DB() - exiting due to too many errors
[2018-10-21 18:46:01.465] dbquery(UPDATE counters SET value = value + 0 WHERE id = 0;) - SQL error (8): attempt to write a readonly database

**Steps to reproduce:**

Open the Long Term Data Page
Select a date range, Last Week for example.
The table never populates.

**Debug token provided by [uploading `pihole -d` log](https://discourse.pi-hole.net/t/the-pihole-command-with-examples/738#debug):**

`ezfjhgnfz8

**Troubleshooting undertaken, and/or other relevant information:**

I'm new to the pi-hole and reading the Discourse forums I found this topic:
https://discourse.pi-hole.net/t/long-term-data-empty-on-v4-0/13605/15

I have two Pis with basic pi-hole installs and both were exhibiting this problem.

User Mcat12 posted a suggestion to change the ownership of the database:

sudo service pihole-FTL stop
sudo chown root:pihole /etc/pihole/pihole-FTL.db
sudo service pihole-FTL start

In addition to the chown above I ended up trying a chmod to 664 as the file was listed as 644 not giving the pihole user write permission. It seems to be working now.

(before) pi@pi-3b-3:~ $ ls -al  /etc/pihole/pihole-FTL.db
-rw-r--r-- 1 root pihole 37810176 Oct 21 17:08 /etc/pihole/pihole-FTL.db

sudo chmod 664 /etc/pihole/pihole-FTL.db

(after) pi@pi-3b-3:~ $ ls -al  /etc/pihole/pihole-FTL.db
-rw-rw-r-- 1 root pihole 37810176 Oct 21 17:08 /etc/pihole/pihole-FTL.db

**Comment by Mcat12** [comment]
Are you still having the issue? The reason it happened was most likely because your device wasn't able to use linux capabilities and had to run FTL as root.

**Comment by stan-qaz** [comment]
TL;DR A new install on new Pi today doesn't have this issue. Closing bug.

I don't think that lacking Linux capabilities was the issue, I had the same issue on two Pi systems both fresh installs of the OS and Pi Hole on Sept 19 2018:

cat /etc/os-release
PRETTY_NAME="Raspbian GNU/Linux 9 (stretch)"
NAME="Raspbian GNU/Linux"
VERSION_ID="9"
VERSION="9 (stretch)"

The command history for both systems still goes back to the original install of pihole and I didn't do anything silly like run the curl installer command as sudo. But still the /etc/pihole/piholeFTL.db on both was owned by root:root and had 644 permissions.

Doing another install on a third Pi just now, and using the same command as before. My Pi is showing that the /etc/pihole/piholeFTL.db is owned by pihole:pihole and had 644 permissions.

No clue what happened but it apparently can't be reproduced today so I'm closing this bug.

BIGFAT in the topic I linked to is apparently having a different issue with his Ubuntu as the suggestion didn't work for him.
https://discourse.pi-hole.net/t/long-term-data-empty-on-v4-0/13605/11
